src/index.ts

```diff
@@ -70,6 +70,8 @@ const program = new Command(packageJson.name)
             'Azure environments: AzureCloud or AzureChinaCloud',
             /^(AzureCloud|AzureChinaCloud)$/i, 'AzureCloud')
     .option('-r, --runtime <runtime>', 'Microservices runtime: dotnet or java', /^(dotnet|java)$/i, 'dotnet')
+    .option('-p, --servicePrincipalId <servicePrincipalId>', 'Service Principal Id')
```

Comment:
nit: for uncommon features, I would avoid the short form, which could be useful for more common scenarios, e.g. "-p" for "port", "print", "password", "path" etc.

Reply:
also when passing a principal id one will have to pass the secret too, for which the short form is not available